lib/GNUMakefilePyGeNN

```diff
@@ -0,0 +1,136 @@
+##--------------------------------------------------------------------------
+##   Author: Thomas Nowotny
+##           James Knight
+##           Anton Komissarov
+##
+##   Institute: Center for Computational Neuroscience and Robotics
+##              University of Sussex
+##              Falmer, Brighton BN1 9QJ, UK
+##
+##   email to:  T.Nowotny@sussex.ac.uk
+##
+##   initial version: 2018-05-14
+##
+##--------------------------------------------------------------------------
+
+SWIG                               :=swig -c++ -python
+SWIG_MODULE_NAME                   :=pygenn
+SWIG_PATH                          :=$(GENN_PATH)/lib/swig
+PYGENN_PATH                        :=$(GENN_PATH)/lib/pygenn
+PYGENN_SWIG_FILE_PATH              :=$(GENN_PATH)/lib/$(SWIG_MODULE_NAME).i
+PYTHON_INCLUDE                     ?=$(shell python-config --includes)
+PYTHON_NUMPY_INCLUDE               ?=$(shell find /usr/lib -type d -path "*/site-packages/numpy/core/include" | grep python3)
+
+# Check whether python includes are defined
+ifeq ($(strip $(PYTHON_INCLUDE)),)
+$(error attempt to automatically detect python include path failed. Try specifying the path as PYTHON_INCLUDE=)
+endif
+ifeq ($(strip $(PYTHON_NUMPY_INCLUDE)),)
+$(error attempt to automatically detect python numpy package include path failed. Try specifying the path as PYTHON_NUMPY_INCLUDE=)
+endif
+# PYGENN link flags
+ifndef CPU_ONLY
+    ifeq ($(DARWIN),DARWIN)
+        PYGENN_LINK_FLAGS       =-rpath $(CUDA_PATH)/lib -L"$(CUDA_PATH)/lib" -lcuda -lcudart -lstdc++ -lc++
+    else
+        ifeq ($(OS_SIZE),32)
+            PYGENN_LINK_FLAGS   =-L"$(CUDA_PATH)/lib" -lcuda -lcudart
+        else
+            PYGENN_LINK_FLAGS   =-L"$(CUDA_PATH)/lib64" -lcuda -lcudart
+        endif
+    endif
+else
+    SWIG_CPU                    :=-DCPU_ONLY
+    PYGENN_PATH                 :=$(addsuffix _CPU,$(PYGENN_PATH))
+    PYGENN_LINK_FLAGS           :=
+    ifeq ($(DARWIN),DARWIN)
+        PYGENN_LINK_FLAGS       :=-lstdc++ -lc++
+    endif
+endif
+
+$(info $$PYGENN_PATH = ${PYGENN_PATH})
+
+ifdef MPI_ENABLE
+    PYGENN_LINK_FLAGS           +=$(shell mpiCC -showme:link)
+endif
+
+PYGENN_PIC_OBJ                     :=global modelSpec neuronGroup synapseGroup currentSource initVarSnippet utils codeStream codeGenUtils sparseUtils neuronModels postSynapseModels synapseModels newNeuronModels newPostsynapticModels newWeightUpdateModels currentSourceModels standardSubstitutions standardGeneratedSections generateALL generateCPU generateInit generateKernels generateMPI generateRunner
```

Comment:
If you included GNUMakefileLibGeNN as I suggest below, I'm **fairly** sure this is just a copy of ``LIBGENN_OBJ``. It would be **REALLY** good if we didn't have to maintain that list in more places (having it in the Windows and *nix build system is bad enough)

Reply:
it is a copy of original `LIBGENN_OBJ` however the `LIBGENN_OBJ` are already fully qualified paths by this point.

Reply:
Yeah I'd be more than happy if you made GNUMakefileLibGeNN more flexible i.e. by keeping the unqualified paths separate.

Reply:
Another issue with this makefile structure that, when you try and do something else with GeNN and you it can't find your python/numpy source, you can't run genn-build-model.sh!